### PR TITLE
AAB packages + SDK 30 + Signature v2 for APKs

### PR DESCRIPTION
**Pull request by nirvn**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f037531f8e2ca78239a1ec2de6bd532645cf0c7d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-2936128a3bf1745af7ce513362d0813275f3f935-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-2936128a3bf1745af7ce513362d0813275f3f935-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-2936128a3bf1745af7ce513362d0813275f3f935-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-2936128a3bf1745af7ce513362d0813275f3f935-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 14d646c3935e8d811bb2394f8ed9ad6b3398fb93:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-a01e343b3e5a9ac5af6b7389f6061d13d27e8e8f-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-a01e343b3e5a9ac5af6b7389f6061d13d27e8e8f-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-a01e343b3e5a9ac5af6b7389f6061d13d27e8e8f-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-a01e343b3e5a9ac5af6b7389f6061d13d27e8e8f-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 78a1238f8bc57c0f65d396a52c79242727476f64:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-5350a6d8da627187d652bdea4a658b1ff94b5efa-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-5350a6d8da627187d652bdea4a658b1ff94b5efa-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-5350a6d8da627187d652bdea4a658b1ff94b5efa-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-5350a6d8da627187d652bdea4a658b1ff94b5efa-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for fa97e8160b3a34cd6f92adb131a9ac8df8b26be9:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-98ffcc18d606223991e0d32958f1a099e998c854-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-98ffcc18d606223991e0d32958f1a099e998c854-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-98ffcc18d606223991e0d32958f1a099e998c854-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-98ffcc18d606223991e0d32958f1a099e998c854-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 158469634406e4df022cba9b22f89a79784948bc:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-0362fec91d9776f42c51e7d084c7aafbb9e69587-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-0362fec91d9776f42c51e7d084c7aafbb9e69587-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-0362fec91d9776f42c51e7d084c7aafbb9e69587-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-0362fec91d9776f42c51e7d084c7aafbb9e69587-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d529d3f789e125f3919073ce0551dd2fc9e169c8:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-5f8a76f9df109fb581b77af7f3c5fc13147b660d-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-5f8a76f9df109fb581b77af7f3c5fc13147b660d-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-5f8a76f9df109fb581b77af7f3c5fc13147b660d-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-5f8a76f9df109fb581b77af7f3c5fc13147b660d-x86.apk)

**Comment by nirvn** [comment]
For the record, to move to target API 30, we'll need to sign APKs using the Signature Scheme v2. I tried earlier and the current signing process leads to bad APKs. Looking at the error message (`InstallFailed (has extras)`) lead me to discover that factoid.

**Comment by nirvn** [comment]
I think this means we can't rely on Qt's androiddeployqt to sign APKs, and instead have to use https://developer.android.com/studio/command-line/apksigner .

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e26036e820740f2caeeea5d4527660f8413e29b2:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-13d5cafcefc9ed30f1b96436901087d3ad8760cc-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-13d5cafcefc9ed30f1b96436901087d3ad8760cc-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-13d5cafcefc9ed30f1b96436901087d3ad8760cc-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-13d5cafcefc9ed30f1b96436901087d3ad8760cc-x86.apk)

**Comment by nirvn** [comment]
This is the error we currently get:

```
adb: failed to install qfield-dev-2288-13d5cafcefc9ed30f1b96436901087d3ad8760cc-arm64_v8a.apk: Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES: Scanning Failed.: No signature found in package of version 2 or newer for package ch.opengis.qfield_beta]
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e26036e820740f2caeeea5d4527660f8413e29b2:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-8a9b6959f96e1e90f538c97fe2223d1afc68a8d1-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-8a9b6959f96e1e90f538c97fe2223d1afc68a8d1-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-8a9b6959f96e1e90f538c97fe2223d1afc68a8d1-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-8a9b6959f96e1e90f538c97fe2223d1afc68a8d1-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 486287f8a3d18039666819c7dec28494513cb58f:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-285d168971d678aa940626421425145162b76609-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-285d168971d678aa940626421425145162b76609-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-285d168971d678aa940626421425145162b76609-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-285d168971d678aa940626421425145162b76609-x86.apk)

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -23,8 +23,8 @@ jobs:
           - triplet: 'x86-android'
 
     env:
-      ANDROID_PLATFORM: 21
-      ANDROID_TARGET_PLATFORM: 29
+      ANDROID_PLATFORM: 24
```
Comment (m-kuhn) [inline .github/workflows/android.yml:26]: Why is that?

Comment (nirvn) [reply .github/workflows/android.yml:26]: @m-kuhn , I couldn't get signature v2 (a requirement when targeting 30) to apply without raising minimum to 24.

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -229,12 +229,14 @@ jobs:
           mv ${{ env.CMAKE_BUILD_DIR }}/android_deployment_settings.json ${{ env.CMAKE_BUILD_DIR }}/android_deployment_settings.tmp
           cat <<< "$(jq ". += { \"sdkBuildToolsRevision\" : \"29.0.2\" }" < ${{ env.CMAKE_BUILD_DIR }}/android_deployment_settings.tmp)" > ${{ env.CMAKE_BUILD_DIR }}/android_deployment_settings.json
           cmake --build  "${{ env.CMAKE_BUILD_DIR }}" --target bundle --config Release
+          /usr/local/lib/android/sdk/build-tools/29.0.2/apksigner sign --ks ./keystore.p12 --ks-pass pass:"${{ secrets.STOREPASS }}" --ks-key-alias "qfield" --key-pass pass:"${{ secrets.KEYPASS }}" ${{ env.CMAKE_BUILD_DIR }}/android-build/build/outputs/apk/release/android-build-release-signed.apk
```
Comment (m-kuhn) [inline .github/workflows/android.yml:232]: Can we get the path from $ENV{ANDROID_NDK_HOME,} ?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 95fecbc87bd4c786d87e80c209ec7541b013fdf4:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-9bc6d55d4a5e3e4cf89d2d7dff643fe1ecbdd644-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-9bc6d55d4a5e3e4cf89d2d7dff643fe1ecbdd644-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-9bc6d55d4a5e3e4cf89d2d7dff643fe1ecbdd644-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-9bc6d55d4a5e3e4cf89d2d7dff643fe1ecbdd644-x86.apk)

File: scripts/ci/upload_artifacts.sh [hunk 3]
```diff
@@ -9,16 +9,25 @@ then
...
-    FILENAME="qfield-${CI_TAG}-${ARCH}.apk"
+    FILENAME_AAB="qfield-${CI_TAG}-${ARCH}.aab"
+    FILENAME_APK="qfield-${CI_TAG}-${ARCH}.apk"
   else
-    FILENAME="qfield-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${ARCH}.apk"
+    FILENAME_AAB="qfield-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${ARCH}.aab"
+    FILENAME_APK="qfield-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${ARCH}.apk"
   fi
 
-  mv ${CMAKE_BUILD_DIR}/android-build/build/outputs/apk/release/android-build-release-signed.apk /tmp/${FILENAME}
```
Comment (m-kuhn) [inline scripts/ci/upload_artifacts.sh:17]: Do we need to preserve this line? Something is still done with it later

**Review by m-kuhn — COMMENTED** [review]
Looks good

File: .github/workflows/android.yml [hunk 4]
```diff
@@ -253,6 +255,12 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
           release-tag: ${{ env.CI_TAG }}
 
+      - name: 🦨 Setup tmate session
```
Comment (m-kuhn) [inline .github/workflows/android.yml:258]: Remove this before final merge

File: scripts/ci/upload_artifacts.sh (hunk 3 above)
Comment (nirvn) [reply scripts/ci/upload_artifacts.sh:17]: @m-kuhn , it's there, a few lines below (line 26)

**Review by m-kuhn — COMMENTED** [review]
One more thing

File: scripts/ci/upload_artifacts.sh (hunk 3 above)
Comment (m-kuhn) [reply scripts/ci/upload_artifacts.sh:17]: Ah, missed that

File: .github/workflows/android.yml [hunk 5]
```diff
@@ -229,7 +229,7 @@ jobs:
           mv ${{ env.CMAKE_BUILD_DIR }}/android_deployment_settings.json ${{ env.CMAKE_BUILD_DIR }}/android_deployment_settings.tmp
           cat <<< "$(jq ". += { \"sdkBuildToolsRevision\" : \"29.0.2\" }" < ${{ env.CMAKE_BUILD_DIR }}/android_deployment_settings.tmp)" > ${{ env.CMAKE_BUILD_DIR }}/android_deployment_settings.json
           cmake --build  "${{ env.CMAKE_BUILD_DIR }}" --target bundle --config Release
-          /usr/local/lib/android/sdk/build-tools/29.0.2/apksigner sign --v2-signing-enabled true --ks ./keystore.p12 --ks-pass pass:"${{ secrets.STOREPASS }}" --ks-key-alias "qfield" --key-pass pass:"${{ secrets.KEYPASS }}" ${{ env.CMAKE_BUILD_DIR }}/android-build/build/outputs/apk/release/android-build-release-signed.apk
+          ${ANDROID_NDK_HOME}/build-tools/29.0.2/apksigner sign --v2-signing-enabled true --ks ./keystore.p12 --ks-pass pass:"${{ secrets.STOREPASS }}" --ks-key-alias "qfield" --key-pass pass:"${{ secrets.KEYPASS }}" ${{ env.CMAKE_BUILD_DIR }}/android-build/build/outputs/apk/release/android-build-release-signed.apk
```
Comment (m-kuhn) [inline .github/workflows/android.yml:232]: I like the ${{ env.ANDROID_NDK_HOME }} notation.
 that will make the path show up in plain text in the logs and makes it easier to debug

**Review by m-kuhn — COMMENTED** [review]
.

**Review by m-kuhn — APPROVED** [review]

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d453703e52ef330befee17f04b14abd8e760679d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-56e3f333620d39b8dc7dcd530a780470b8717766-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-56e3f333620d39b8dc7dcd530a780470b8717766-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-56e3f333620d39b8dc7dcd530a780470b8717766-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-56e3f333620d39b8dc7dcd530a780470b8717766-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b4264cfe73b92733fdda65626902cee920997be0:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-f7b4fd7ff58a394a37a03060c995a4644e4014b9-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-f7b4fd7ff58a394a37a03060c995a4644e4014b9-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-f7b4fd7ff58a394a37a03060c995a4644e4014b9-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-f7b4fd7ff58a394a37a03060c995a4644e4014b9-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f8696fdb16e0599b23bf6cb8312f8a5ce953f9b3:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-774b1ee3c4bd29b8cd9dbd77038cf8f975fc95b4-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-774b1ee3c4bd29b8cd9dbd77038cf8f975fc95b4-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-774b1ee3c4bd29b8cd9dbd77038cf8f975fc95b4-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2288-774b1ee3c4bd29b8cd9dbd77038cf8f975fc95b4-x86.apk)

**Pull request merged**